File: backend/app/src/main/java/com/yagubogu/place/dto/v1/PlaceDetailResponse.java
package com.yagubogu.place.dto.v1;

import com.yagubogu.place.domain.Place;
import com.yagubogu.place.domain.PlaceCategory;
import com.yagubogu.place.dto.v1.detail.PlaceDetail;

public record PlaceDetailResponse(
        Long id,
        PlaceCategory category,
        String title,
        String address,
        Double mapX,
        Double mapY,
        String tel,
        String imageUrl,
        String overview,
        String homepage,
        PlaceDetail detail
) {
    public static PlaceDetailResponse from(Place place, String overview, String homepage, PlaceDetail detail) {
        return new PlaceDetailResponse(
                place.getId(),
                place.getCategory(),
                place.getTitle(),
                place.getAddress(),
                place.getMapX(),
                place.getMapY(),
                place.getTel(),
                place.getImageUrl(),
                overview,
                homepage,
                detail
        );
    }
}

File: mobile/composeApp/src/commonMain/kotlin/com/yagubogu/data/dto/response/place/PlaceDetailResponse.kt
package com.yagubogu.data.dto.response.place

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

@Serializable
data class PlaceDetailResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("category")
    val category: PlaceCategoryDto,
    @SerialName("title")
    val title: String,
    @SerialName("address")
    val address: String? = null,
    @SerialName("mapX")
    val mapX: Double,
    @SerialName("mapY")
    val mapY: Double,
    @SerialName("tel")
    val tel: String? = null,
    @SerialName("imageUrl")
    val imageUrl: String? = null,
    @SerialName("overview")
    val overview: String? = null,
    @SerialName("homepage")
    val homepage: String? = null,
    @SerialName("detail")
    val detail: JsonObject? = null,
)
